webapp/sources/rudder/rudder-web/src/main/scala/com/normation/plugins/RudderPluginJson.scala

```diff
@@ -0,0 +1,131 @@
+/*
+*************************************************************************************
+* Copyright 2019 Normation SAS
+*************************************************************************************
+*
+* This file is part of Rudder.
+*
+* Rudder is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* In accordance with the terms of section 7 (7. Additional Terms.) of
+* the GNU General Public License version 3, the copyright holders add
+* the following Additional permissions:
+* Notwithstanding to the terms of section 5 (5. Conveying Modified Source
+* Versions) and 6 (6. Conveying Non-Source Forms.) of the GNU General
+* Public License version 3, when you create a Related Module, this
+* Related Module is not considered as a part of the work and may be
+* distributed under the license agreement of your choice.
+* A "Related Module" means a set of sources files including their
+* documentation that, without modification of the Source Code, enables
+* supplementary functions or services in addition to those offered by
+* the Software.
+*
+* Rudder is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+*
+*************************************************************************************
+*/
+
+package com.normation.plugins
+
+import java.nio.charset.StandardCharsets
+
+import better.files.File
+import org.joda.time.DateTime
+import zio._
+import zio.syntax._
+import com.normation.errors._
+import net.liftweb.json._
+import org.joda.time.format.ISODateTimeFormat
+
+/*
+ * This file is able to read information from /var/rudder/packages/index.json
+ * (package management file for plugins)
+ */
+
+// this is only used internally but liftweb can't hand it if not top level
+protected final case class JsonPluginFile(plugins: Map[String, JValue])
+
+// this is only used internally but liftweb can't hand it if not top level
+protected final case class JsonPluginRaw(
+    name          : String
+  , version       : String
+  , files         : List[String]
+  , `jar-files`   : List[String]
```

Comment:
why the ` ` ?

Reply:
The? Backquote? It's to allow `-` in the name.

Reply:
oh, ok